src/otx/core/metrics/accuracy.py

```diff
@@ -5,16 +5,198 @@
+
+        msg = f"Average should be MICRO or MACRO, got {self.average}"
+        raise ValueError(msg)
+
+    def compute(self) -> Tensor | dict[str, Any]:
+        """Compute the metric."""
+        conf_matrices = self._compute_unnormalized_confusion_matrices()
+
+        return {
+            "conf_matrix": conf_matrices,
+            "accuracy": self._compute_accuracy_from_conf_matrices(conf_matrices),
+        }
+
+
+class CustomMulticlassAccuracy(CustomAccuracy):
+    """Custom accuracy class for the multi-class classification."""
+
+    def _compute_unnormalized_confusion_matrices(self) -> list[NamedConfusionMatrix]:
+        """Compute an unnormalized confusion matrix for every label group."""
+        conf_matrices = []
+        for label_group in self.label_info.label_groups:
+            label_to_idx = {label: index for index, label in enumerate(self.label_info.label_names)}
+            group_indices = [label_to_idx[label] for label in label_group]
+
+            mask = torch.tensor([t.item() in group_indices for t in self.targets])
+            valid_preds = torch.tensor(self.preds)[mask]
+            valid_targets = torch.tensor(self.targets)[mask]
+
+            for i, index in enumerate(group_indices):
+                valid_preds[valid_preds == index] = i
+                valid_targets[valid_targets == index] = i
+
+            num_classes = len(label_group)
+            confmat = NamedConfusionMatrix(
+                task="multiclass",
+                num_classes=num_classes,
+                row_names=label_group,
+                col_names=label_group,
+            )
+            conf_matrices.append(confmat(valid_preds, valid_targets))
+        return conf_matrices
+
+
+class CustomMultilabelAccuracy(CustomAccuracy):
+    """Custom accuracy class for the multi-label classification."""
+
+    def _compute_unnormalized_confusion_matrices(self) -> list[NamedConfusionMatrix]:
+        """Compute an unnormalized confusion matrix for every label group."""
+        preds = torch.stack(self.preds)
+        targets = torch.stack(self.targets)
+
+        conf_matrices = []
+        for i, label_group in enumerate(self.label_info.label_groups):
+            label_preds = (preds[:, i] >= self.threshold).long()
+            label_targets = targets[:, i]
+
+            valid_mask = label_targets >= 0
+            if valid_mask.any():
+                valid_preds = label_preds[valid_mask]
+                valid_targets = label_targets[valid_mask]
+            else:
+                continue
+
+            data_name = [label_group[0], "~" + label_group[0]]
+            confmat = NamedConfusionMatrix(task="binary", num_classes=2, row_names=data_name, col_names=data_name).to(
+                self.device,
+            )
+            conf_matrices.append(confmat(valid_preds, valid_targets))
+        return conf_matrices
+
+
+class CustomHlabelAccuracy(CustomAccuracy):
```

Comment:
Actually, there is no traditional HLabelAccuracy. I added docstring what it really do. https://github.com/openvinotoolkit/training_extensions/pull/2994/commits/680ccc68aee3ef0d9d051a54517d348a867a2849

Reply:
Can we rename this to `HLabelAccuracy`

Reply:
We always open to change the name, however, for this case, I think `CustomHLabelAccuracy` is better since all other metrics including this are inherited from the `CustomAccuracy` and this comes from the different behavior that averages through the label_groups.